### PR TITLE
get the api connection handle for provider with mgmt_system

### PR DESCRIPTION
**Pull request by amavinag**
* the get_mgmt(provider).api is added
  * added method 'is_pingable(ip)' to check if provider IP
    is reachable, if not reachable then 'continue' to next
    available provider.
  * changed the edomain maintenance status, as edomain needs
    to be in active state to access api data.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit f29e06c15042e1842840f1ddd6dc45cda809e118:

`scripts/cleanup_edomain_templates.py`:
- :bangbang: Line 204:101: **E501** *line too long (101 > 100 characters)*
- :grey_exclamation: Line 296:13: **T001** *print statement found.*
- :bangbang: Line 304:19: **E201** *whitespace after '('*

`scripts/template_upload_rhevm.py`:
- :grey_exclamation: Line 229:9: **T001** *print statement found.*
- :grey_exclamation: Line 317:13: **T001** *print statement found.*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 702641e8c7dcbecce74eaa4dfc027ff92b6125e3:

`scripts/cleanup_edomain_templates.py`:
- File lint OK :cake: :punch: :cookie:

`scripts/template_upload_rhevm.py`:
- :grey_exclamation: Line 316:13: **T001** *print statement found.*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f2078d96f64c1ff10ea0b91006ea49f872e1d51a:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8542e1eae17ebdf5a774c9d3f0568a382fe4cc94:

Everything seems all right :smile: .
*CFME QE Bot*

File: scripts/cleanup_edomain_templates.py [hunk 1]
```diff
@@ -49,6 +50,24 @@ def make_ssh_client(rhevip, sshname, sshpass):
     return SSHClient(**connect_kwargs)
 
 
+def is_pingable(ip_address):
```
Comment (mfalesni) [inline scripts/cleanup_edomain_templates.py]: Couple of thoughts:

* Could this live in `utils.net`? I think we already have some function to check if a port is open, maybe that could be used instead if you know the port on which RHEV communicates.
* The comment has the wrong doc format :) Check out eg. `cfme.web_ui` for the example.

**Comment by dajoRH** [comment]
Lint report for commit daced8bc6859178b2b985c0de127b736a07ede26:

`scripts/cleanup_edomain_templates.py`:
- :red_circle: Line 12:1: **F401** *'os' imported but unused*

`scripts/template_upload_rhevm.py`:
- File lint OK :cake: :punch: :cookie:

`utils/net.py`:
- File lint OK :cake: :punch: :cookie:

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6f5ded551d89d8bae0a44ace6ed49ee77352a237:

Everything seems all right :smile: .
*CFME QE Bot*

File: scripts/template_upload_rhevm.py [hunk 2]
```diff
@@ -78,7 +78,8 @@ def change_edomain_state(api, state, edomain):
                 print '{} successfully set to {} state'.format(edomain, state)
                 return True
         return False
-    except Exception:
+    except Exception as e:
+        print(e)
```
Comment (psav) [inline scripts/template_upload_rhevm.py]: If we are changing prints, there is one below to change here too

**Comment by psav** [comment]
I think I'm happy here, @mfalesni ?

File: utils/net.py [hunk 3]
```diff
@@ -133,3 +134,22 @@ def resolve_ips(host_iterable, force_dns=False):
...
+    try:
+        status = os.system("ping -c1 -w2 {}".format(ip_addr))
+        if status == 0:
+            print('IP: {} is UP !'.format(ip_addr))
+            return True
+        print('IP: {} is DOWN !'.format(ip_addr))
+        return False
+    except Exception as e:
+        print(e)
+        return False
```
Comment (mfalesni) [inline utils/net.py]: @psav @amavinag I would say that if it is not inside the script, it should only output into a logger. All three messages, first two should use logger.info, the exceptions shoulddo logger.exception(e).

**Comment by mfalesni** [comment]
Apart from the log-prints in utils/net, I think it is good. I would appreciate if the except/return False printed out the exception, but that is not a big fuss.

**Comment by dajoRH** [comment]
Lint report for commit c1778cb213e8255b47f87e3e36672de3e66e63fd:

`scripts/template_upload_rhevm.py`:
- :bangbang: Line 81:27: **E901** *SyntaxError: invalid syntax*
- :bangbang: Line 82:9: **E113** *unexpected indentation*

`utils/net.py`:
- :bangbang: Line 151:1: **E101** *indentation contains mixed spaces and tabs*
- :heavy_exclamation_mark: Line 151:1: **W191** *indentation contains tabs*
- :bangbang: Line 152:1: **E101** *indentation contains mixed spaces and tabs*
- :bangbang: Line 153:1: **E101** *indentation contains mixed spaces and tabs*
- :heavy_exclamation_mark: Line 153:1: **W191** *indentation contains tabs*
- :bangbang: Line 154:1: **E101** *indentation contains mixed spaces and tabs*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 876d17784980d459b5e38eace5641c5bab70f138:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**